src/util/mpir_pmi.c

```diff
@@ -264,42 +264,51 @@ int MPIR_pmi_barrier(MPIR_PMI_DOMAIN domain)
 
-    if (domain == MPIR_PMI_DOMAIN_LOCAL) {
-        /* use local proc set */
-        /* FIXME: we could simply construct the proc set from MPIR_Process.node_local_map */
-        pmix_proc_t *procs;
-        size_t nprocs;
-        pmix_value_t *pvalue = NULL;
+    /* use global wildcard proc set */
+    pmi_errno = PMIx_Fence(&pmix_wcproc, 1, info, 1);
+    MPIR_ERR_CHKANDJUMP1(pmi_errno != PMIX_SUCCESS, mpi_errno, MPI_ERR_OTHER,
+                         "**pmix_fence", "**pmix_fence %d", pmi_errno);
+    PMIX_INFO_FREE(info, 1);
+#endif
 
-        pmi_errno = PMIx_Get(&pmix_proc, PMIX_HOSTNAME, NULL, 0, &pvalue);
-        MPIR_ERR_CHKANDJUMP1(pmi_errno != PMIX_SUCCESS, mpi_errno, MPI_ERR_OTHER, "**pmix_get",
-                             "**pmix_get %d", pmi_errno);
-        const char *nodename = (const char *) pvalue->data.string;
+  fn_exit:
+    return mpi_errno;
+  fn_fail:
+    goto fn_exit;
+}
 
-        pmi_errno = PMIx_Resolve_peers(nodename, pmix_proc.nspace, &procs, &nprocs);
-        MPIR_ERR_CHKANDJUMP1(pmi_errno != PMIX_SUCCESS, mpi_errno, MPI_ERR_OTHER,
-                             "**pmix_resolve_peers", "**pmix_resolve_peers %d", pmi_errno);
+int MPIR_pmi_barrier_local(void)
+{
+#if defined(USE_PMIX_API)
+    int mpi_errno = MPI_SUCCESS;
+    /* FIXME: we could simply construct the proc set from MPIR_Process.node_local_map */
+    pmix_proc_t *procs;
+    size_t nprocs;
+    pmix_value_t *pvalue = NULL;
 
-        pmi_errno = PMIx_Fence(procs, nprocs, info, 1);
-        MPIR_ERR_CHKANDJUMP1(pmi_errno != PMIX_SUCCESS, mpi_errno, MPI_ERR_OTHER, "**pmix_fence",
-                             "**pmix_fence %d", pmi_errno);
+    pmi_errno = PMIx_Get(&pmix_proc, PMIX_HOSTNAME, NULL, 0, &pvalue);
+    MPIR_ERR_CHKANDJUMP1(pmi_errno != PMIX_SUCCESS, mpi_errno, MPI_ERR_OTHER, "**pmix_get",
+                         "**pmix_get %d", pmi_errno);
+    const char *nodename = (const char *) pvalue->data.string;
 
-        PMIX_VALUE_RELEASE(pvalue);
-        PMIX_PROC_FREE(procs, nprocs);
+    pmi_errno = PMIx_Resolve_peers(nodename, pmix_proc.nspace, &procs, &nprocs);
+    MPIR_ERR_CHKANDJUMP1(pmi_errno != PMIX_SUCCESS, mpi_errno, MPI_ERR_OTHER,
+                         "**pmix_resolve_peers", "**pmix_resolve_peers %d", pmi_errno);
 
-    } else {
-        /* use global wildcard proc set */
-        pmi_errno = PMIx_Fence(&pmix_wcproc, 1, info, 1);
-        MPIR_ERR_CHKANDJUMP1(pmi_errno != PMIX_SUCCESS, mpi_errno, MPI_ERR_OTHER,
-                             "**pmix_fence", "**pmix_fence %d", pmi_errno);
-    }
-    PMIX_INFO_FREE(info, 1);
-#endif
+    pmi_errno = PMIx_Fence(procs, nprocs, info, 1);
+    MPIR_ERR_CHKANDJUMP1(pmi_errno != PMIX_SUCCESS, mpi_errno, MPI_ERR_OTHER, "**pmix_fence",
+                         "**pmix_fence %d", pmi_errno);
+
+    PMIX_VALUE_RELEASE(pvalue);
+    PMIX_PROC_FREE(procs, nprocs);
 
   fn_exit:
     return mpi_errno;
   fn_fail:
     goto fn_exit;
+#else
```

Comment:
Probably somewhat helpful to put a comment on what this `else` is for. `!USE_PMIX_API` case, I guess?